### PR TITLE
Update to use the kie package structure

### PR DESCRIPTION
**Pull request by nrlucaroni**
As an inexperienced user of Drools myself, I had no idea the previous package format was deprecated a long time ago to this kie package structure. This change-set updates to that package structure and will be compatible with drools >6. Interestingly, the dsl as an additional argument really isn't necessary as we just load in all the files necessary into the kieFileSystem, but I've left it separate for clarity.

I realize you may not be wanting to upgrade yet, but if you want to use Java8 features an upgrade is necessary.

### CODE REVIEW

File: src/main/java/pl/maciejwalkowiak/drools/DroolsInjector.java [hunk 1]
```diff
@@ -39,48 +38,33 @@ public void initDrools(Object testClass) throws Exception {
...
-    private InputStreamReader loadDroolFile(String droolsLocation, String filename) {
+    private void loadDroolFile(KieFileSystem kieFileSystem, String droolsLocation, String filename) {
+        KieResources kieResources = kieServices.getResources();
         InputStream stream = getClass().getResourceAsStream(droolsLocation + filename);
-
         if (stream == null) {
             throw new IllegalArgumentException("File not found in location: " + droolsLocation + filename);
         }
-        return new InputStreamReader(stream);
+        kieFileSystem.write("src/main/resources/" + droolsLocation + filename, kieResources.newInputStreamResource(stream));
```
Comment (nrlucaroni) [inline src/main/java/pl/maciejwalkowiak/drools/DroolsInjector.java]: I'm not a fan of this directly. Apparently all of these drools files must be loaded into the kie filesystem with path starting with src/main/resources; maybe you have an idea of how to do this a little more appropriately.

File: pom.xml [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
-    <version>1.1-SNAPSHOT</version>
+    <version>2.0-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <name>junit-drools</name>
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <drools.version>6.0.1.Final</drools.version>
+        <drools.version>6.2.0.CR2</drools.version>
```
Comment (nrlucaroni) [inline pom.xml:14]: this number could be reduced (I believe the minimum is version 5), but the old stuff was removed in 6.1 .

File: src/main/java/pl/maciejwalkowiak/drools/DroolsAnnotationProcessor.java [hunk 3]
```diff
@@ -23,17 +26,24 @@ public DroolsFiles getDroolsFiles() {
...
+        while (klass != null) {
+            allFields.addAll(Arrays.asList(klass.getDeclaredFields()));
+            klass = klass.getSuperclass();
+        }
+        return allFields;
+    }
+
     public void setDroolsSession(pl.maciejwalkowiak.drools.DroolsSession droolsSession) {
-        for (Field field : testClass.getClass().getDeclaredFields()) {
+        for (Field field : getAllFields()) {
```
Comment (nrlucaroni) [inline src/main/java/pl/maciejwalkowiak/drools/DroolsAnnotationProcessor.java]: we've been using a general class to load in globals and attach some custom listeners, but also adding the droolssession annotation and init method would be a big help and this allows that.

**Comment by maciejwalkowiak** [comment]
Unfortunately due to the lack of time I am not able to continue developing and maintaining `junit-drools` project. Thank you for your contributions and I hope you will find one of the forks still maintained and helpful for you!

**Pull request closed without merging**